src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -96,6 +100,19 @@ private void setSubmissionData(Map<String, Object> data, String overseasEntityId
         data.put(MANAGING_OFFICERS_INDIVIDUAL_FIELD, submissionDto.getManagingOfficersIndividual());
         data.put(MANAGING_OFFICERS_CORPORATE_FIELD, submissionDto.getManagingOfficersCorporate());
         data.put(BENEFICIAL_OWNERS_STATEMENT, submissionDto.getBeneficialOwnersStatement());
+
+        if (submissionDto.getTrusts() != null) {
+            // Convert trust data to JSON string if it exists on transaction else set it to an empty string
+            ObjectMapper mapper = JsonMapper.builder().findAndAddModules().build();
+            try {
+                data.put(TRUST_DATA, mapper.writeValueAsString(submissionDto.getTrusts()));
+            } catch (JsonProcessingException e) {
+                throw new ServiceException("Error converting trust data to JSON " + e.getMessage(), e);
+            }
+        } else {
```

Comment:
I'd write this slightly differently, but up to you... Would define a trustData string variable before the if block and initialise it as an empty string - code in if block would overwrite its value. The else block can then go. And you'd only have one data.put() line immediately after the if block that sets the 'trustData' variable on the data map.